src/controllers/publish.ts

```diff
@@ -295,10 +279,223 @@ export const taskList = async (req: Request, res: Response, next: NextFunction)
+                logger.debug('Sending file to backend.');
+                await req.swapi.uploadMeasureLookup(dataset.id, fileData, fileName);
+                res.redirect(req.buildUrl(`/publish/${dataset.id}/measure/review`, req.language));
+                return;
+            } catch (err) {
+                const error = err as ApiException;
+                logger.debug(`Error is: ${JSON.stringify(error, null, 2)}`);
+                if (error.status === 400) {
+                    logger.error('Measure lookup table did not match data in the fact table.', err);
+                    const failurePreview = JSON.parse(error.body as string) as ViewErrDTO;
+                    res.status(400);
+                    res.render('publish/measure-match-failure', {
+                        ...failurePreview,
+                        measure
+                    });
+                    return;
+                }
+                logger.error('Something went wrong other than not matching');
+                logger.debug(`Full error JSON: ${JSON.stringify(error, null, 2)}`);
+                req.session.errors = [
+                    {
+                        field: 'unknown',
+                        message: {
+                            key: 'errors.csv.unknown'
+                        }
+                    }
+                ];
+                res.redirect(req.buildUrl(`/publish/${dataset.id}/measure`, req.language));
+                return;
+            }
+        }
+
+        const dataPreview = await req.swapi.getMeasurePreview(res.locals.dataset.id);
+        if (req.session.errors) {
+            const errors = req.session.errors;
+            req.session.errors = undefined;
+            req.session.save();
+            res.status(500);
+            res.render('publish/measure-preview', {
+                ...dataPreview,
+                measure,
+                errors
+            });
+        } else {
+            res.render('publish/measure-preview', { ...dataPreview, measure });
+        }
+    } catch (err) {
+        logger.error('Failed to get dimension preview', err);
+        next(new NotFoundException());
+    }
+};
+
+export const measureReview = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const dataset = singleLangDataset(res.locals.dataset, req.language);
+        const measure = dataset.measure;
+        if (!measure) {
+            logger.error('Failed to find measure in dataset');
+            next(new NotFoundException());
+            return;
+        }
+        let errors: ViewErrDTO | undefined;
+
+        if (req.method === 'POST') {
+            logger.debug(`User has reviewed measure lookup table.`);
+            switch (req.body.confirm) {
+                case 'true':
+                    res.redirect(req.buildUrl(`/publish/${dataset.id}/tasklist`, req.language));
+                    break;
+                case 'goback':
```

Comment:
Tiny thing but just for consistency's sake I'd suggest either `true` & `false` or maybe `continue` & `cancel`.

Reply:
Changed to `continue` and `cancel`